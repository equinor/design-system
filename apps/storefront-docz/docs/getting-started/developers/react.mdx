---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
