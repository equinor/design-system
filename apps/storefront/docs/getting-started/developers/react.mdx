---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
