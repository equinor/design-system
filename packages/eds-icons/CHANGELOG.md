# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.18.0] - 2023-01-26

### Added

- `stop_circle`
- `stop_circle_outlined`

### Fixed

- Rendering error `line`
- Rendering error `onshore_drone`

## [0.17.0] - 2022-11-25

### Added

- `wellbore`

## [0.16.0] - 2022-11-07

### Added

- `enlarge`
- `offline_document`
- `open_side_sheet`
- `reduce`
- `sheet_bottom_position`
- `sheet_leftposition`

### Fixed

- Rendering error `approve`
- Rendering error `tag_more`

### Changed

- Resized `in_progress`

## [0.15.0] - 2022-10-12

### Added

- `in_progress`

### Removed

- 🔥 Remove commonjs path for in eds-icons & eds-tokens by @mimarz in https://github.com/equinor/design-system/pull/2571

### Changed

- 🔧 Conform packages build, test & linting by @mimarz in https://github.com/equinor/design-system/pull/2555

## [0.14.0] - 2022-09-19

### Added

- `filter_alt_active`
- `filter_alt_off`
- `collapse_screen`
- `expand_screen`
- `minimize`
- `slack`

### Changed

- Redesigned arrow on `priority_low` & `priority_high`

## [0.13.0] - 2022-08-30

### Added

- `oil`
- `oil_barrel`

## [0.12.0] - 2022-06-30

## Added

- `signature`
- `approve`
- `no_craning`
- `craning`
- `toolbox`
- `toolbox_wheel`
- `toolbox_rope`
- `badge`
- `hand_radio`
- `substation_onshore`
- `substation_offshore`
- `expand`
- `collapse`

## [0.11.0] - 2022-06-09

## Added

- `zip_file`
- `subsea_drone`
- `onshore_drone`
- `anchor`
- `aerial_drone`

## Fixed

- Fixed typo in icons. Make sure to update the following icons to fixed name
  - `communte` -> `commute`
  - `headseat_mic` -> `headset_mic`
  - `desktop_windwos` -> `desktop_windows`
  - `breifcase` -> `briefcase`
  - `format_stikethrough` -> `format_strikethrough`
  - `mood_extreamly_sad` -> `mood_extremely_sad`
  - `mood_extreamly_happy` -> `mood_extremely_happy`

## Removed

- `communte`
- `headseat_mic`
- `desktop_windwos`
- `breifcase`
- `format_stikethrough`
- `mood_extreamly_sad`
- `mood_extreamly_happy`

## [0.10.0] - 2021-12-20

## Added

- `wind_turbine` icon ([#1801](https://github.com/equinor/design-system/issues/1801))

## Fixed

- `calendar` icon should no longer misaligned with other "calendar" icons ([#1809](https://github.com/equinor/design-system/issues/1809))

## [0.9.1] - 2021-12-10

## Fixed

- Missing vector on `fault` icon ([#1793](https://github.com/equinor/design-system/issues/1793))

## [0.9.0] - 2021-12-09

## Added

New icons ([#1743](https://github.com/equinor/design-system/issues/1743)) ([#1581](https://github.com/equinor/design-system/issues/1581)) ([#1734](https://github.com/equinor/design-system/issues/1734))

- `ducting`
- `electrical`
- `fault`
- `formula`
- `go_to`
- `grid_layer`
- `grid_layers`
- `heat_trace`
- `hill_shading`
- `instrument`
- `junction_box`
- `line`
- `log_in`
- `log_out`
- `manual_valve`
- `miniplayer_fullscreen`
- `miniplayer`
- `pipe_support`
- `surface_layer`
- `tag_main_equipment`
- `tag_more`
- `tag_relations`
- `tag_special_equipment`
- `telecom`
- `thumb_pin`
- `well`

## Changed

- `launch` (should no longer be confused with `external_link` icon) ([#1742](https://github.com/equinor/design-system/issues/1742))
- Renamed `inspectrotation` to `inspect_rotation` ([#1781](https://github.com/equinor/design-system/issues/1781))
- Renamed `inspect3d` to `inspect_3d` ([#1781](https://github.com/equinor/design-system/issues/1781))
- Renamed `rotate3d` to `rotate_3d` ([#1781](https://github.com/equinor/design-system/issues/1781))

## [0.8.0] - 2021-11-09

### Deprecated

- Removed `umd` module from the build. This should prompt build tools select the tree-shakeable `esm` module instead. ([#1539](https://github.com/equinor/design-system/issues/1539))

## [0.7.0] - 2021-09-07

### Added

- New icons:
  - `power_bi`
  - `engineering`
  - `boundaries`
  - `inspect3d`
  - `inspectrotation`
  - `invert`

### Changed

- Updated `svgo` from `1.3.2` to `2.3.0` with new parsing, bundling and minimizing plugins.

## [0.6.2] - 2021-01-21

### Fixed 🐛

- Icon name typo for icon `close_cricle_outlined ` -> `close_circle_outlined ` (#993)

## [0.6.1] - 2020-12-11

### Fixed

- Removed postinstall script

## [0.6.0] - 2020-11-30

### Fixed

- Fixed icons with typo `verticle` -> `vertical`
  - more_verticle
  - verticle_split
  - text_rotation_verticle
  - verticle_align_top
  - verticle_align_center
  - verticle_align_bottom
  - border_verticle
  - swap_verticle
  - swap_verticle_circle

### Deprecated

- `@equinor/eds-icons/commonjs` path will be removed as defined as `commonjs` is the `main` type for the package now.

## [0.5.1] - 2020-11-26

### Changed 📓

- Updated README.md

## [0.5.0] - 2020-11-26

### Added

- Types, as part of the [Typescript Milestone](https://github.com/equinor/design-system/milestone/7?closed=1)

### Changed

- Changed module types for better support with `cjs` and `esm`. Using the `<some-eds-npm-package>/commonjs` path on packages should no longer be needed and will be deprecated in the future ([#887](https://github.com/equinor/design-system/issues/887))

## [0.4.0] - 2020-09-11

### Added

- New icon as alternative to `filter` (funnel) called `filter_alt`

### Changed

- Tweaked svg data for `info_circle` & `list`

## [0.3.0] - 2020-06-09

### Added

- New icon for `cable` under **Technical**

### Fixed

- Fixed missing circle on `info_circle` icon ([#349](https://github.com/equinor/design-system/issues/349))

## [0.2.0] - 2020-05-29

### Added

- New group of icons, **Technical**, including _gear, bearing, pressure, platform, circuit, gas_ & _beat_ icons

## [0.1.0] - 2020-02-24

### Changed

- Updated README to me abit more clearer
- Adjustments to to `package.json` to be more in line with our other packages
- New versioning of packages

## [0.0.1-alpha.3] - 2020-01-24

### Added

- Exports of objects containing svg data
- Support for commonjs under seperate path `@equinor/eds-icons/commonjs`

### Removed

- Svg files
- Exports of full svgs as strings

## [0.0.1-alpha.2] - 2019-11-08

### Changed

- Added rollup builds

## [0.0.1-alpha.1] - 2019-11-08

### Changed

- Publish alpha-version of library
