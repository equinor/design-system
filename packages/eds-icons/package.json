{
  "name": "@equinor/eds-icons",
  "version": "0.21.0",
  "description": "Icons from the Equinor Design System",
  "main": "dist/icons.cjs",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "license": "Apache-2.0",
  "author": {
    "name": "EDS Core Team",
    "email": "fg_eds@equinor.com"
  },
  "homepage": "https://eds.equinor.com",
  "repository": {
    "type": "git",
    "url": "https://github.com/equinor/design-system",
    "directory": "packages/eds-icons"
  },
  "bugs": {
    "url": "https://github.com/equinor/design-system/issues"
  },
  "files": [
    "dist/*",
    "commonjs/*"
  ],
  "scripts": {
    "build": "rollup -c --bundleConfigAsCjs && tsc",
    "dev": "rollup -c -w",
    "types": "tsc"
  },
  "keywords": [
    "eds",
    "design system",
    "equinor",
    "icons"
  ],
  "devDependencies": {
    "@rollup/plugin-babel": "^6.0.4",
    "@rollup/plugin-commonjs": "^27.0.0",
    "@rollup/plugin-node-resolve": "^15.3.0",
    "rollup": "^4.22.5",
    "rollup-plugin-delete": "^2.1.0",
    "typescript": "^5.5.4"
  }
}
