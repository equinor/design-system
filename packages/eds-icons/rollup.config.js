import resolve from '@rollup/plugin-node-resolve'
import pkg from './package.json'
import typescript from 'rollup-plugin-typescript2'
import del from 'rollup-plugin-delete'
import commonjsPkg from './commonjs/package.json'

// eslint-disable-next-line import/no-default-export
export default [
  {
    input: 'index.ts',
    watch: {
      clearScreen: true,
    },
    plugins: [
      del({ targets: 'dist/*', runOnce: true }),
      resolve(),
      typescript({ useTsconfigDeclarationDir: true }),
    ],
    output: [
      {
        file: pkg.module,
        name: pkg.name,
        format: 'es',
        exports: 'named',
      },
      {
        file: commonjsPkg.main.replace('../', ''),
        name: pkg.name,
        format: 'cjs',
        exports: 'named',
      },
    ],
  },
]
