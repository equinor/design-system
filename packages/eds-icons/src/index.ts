export * from './data'
export * from './types'
