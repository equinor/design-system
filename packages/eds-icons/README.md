# @equinor/eds-icons

This package is a collection of the `system icons` from the Equinor Design System as javascript objects.

## Installation

```sh
npm install @equinor/eds-icons 
```
If you use Typescript, make sure you have typescript >= 3.8 as a devDependency:
```sh
npm install typescript --save-dev
```

## Usage

Import svg data:

```javascript
import { save } from "@equinor/eds-icons"
```

### React

We have developed a react component to simplify use of the EDS icons library within react.

```javascript
import { Icon } from '@equinor/eds-core-react'
import { save } from '@equinor/eds-icons'

// Use library approach 
Icon.add({ save }) // (this needs only be done once)
<Icon name="save" />

// OR 

// Use directly via data prop:
<Icon data={save} />

```

### Other web frameworks

You can render the EDS icons using plain [svg](https://developer.mozilla.org/en-US/docs/Web/SVG).

‚ö†Ô∏è Due to how Figma parses and exports svg icons the following attributes must be added to your `<path>` element; `fill-rule="evenodd" clip-rule="evenodd"`.

These are not included to reduce bundle size as they are needed for every icon.

```javascript
import { save } from '@equinor/eds-icons'

<svg viewBox={`0 0 ${save.width} ${save.height}`}>
  <path d={save.svgPathData} fill-rule="evenodd" clip-rule="evenodd">
</svg>
```

#### Icon names

Use [EDS Assets in Figma](https://www.figma.com/file/BQjYMxdSdgRkdhKTDDU7L4KU/Assets?node-id=2%3A3)(Equinor Figma account neededüîí) or our [icon preview in Storybook](https://s478stedsstorybookdev.z16.web.core.windows.net/?path=/docs/icons--preview) for icon names. Spaces in icon names are replaced with underscores. 

`star-filled -> star_filled`.

### Example of icon data

```javascript
{
  name: 'star_filled',
  prefix: 'eds',
  height: '24',
  width: '24',
  svgPathData: 'M12 16.067l4.947 3.6-1.894-5.814L20 10.334h-6.067l-1.933-6-1.933 6H4l4.947 3.52-1.894 5.814 4.947-3.6z',
}
```

### FAQ

* If you are using NodeJS 13+ for module loading, it now has native ES modules support. Add `"type":"module"` to your apps `package.json` to enable this.

* If you get a syntax error trying to import ES module, try importing the CJS module as not all loaders have support for ES modules yet

## Credits

The EDS system icons are built on a copy of the [Outlined Material Design](https://material.io/resources/icons/?style=outline) icons provided open-source by Google. The icons have been customised and renamed for Equinor‚Äôs use.
