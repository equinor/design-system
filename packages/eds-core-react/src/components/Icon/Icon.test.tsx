/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { save } from '@equinor/eds-icons'
import { Icon } from '.'

Icon.add({ save })

const StyledIcon = styled(Icon)`
  position: relative;
  height: 100px;
  width: 100px;
`

describe('Icon', () => {
  it('Matches snapshot using data', () => {
    const { asFragment } = render(<Icon data={save} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test using data', async () => {
    const { container } = render(<Icon data={save} />)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Should pass a11y test using name', async () => {
    const { container } = render(<Icon name="save" />)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Should pass a11y test using title', async () => {
    const title = 'Save me'
    const { container } = render(<Icon name="save" title={title} />)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Matches snapshot using name ', () => {
    const { asFragment } = render(<Icon name="save" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Has correct svg data', () => {
    render(<Icon name="save" />)
    expect(screen.queryByTestId('eds-icon-path')).toHaveAttribute(
      'd',
      save.svgPathData,
    )
  })
  it('Has correct svg data when using data property', () => {
    render(<Icon data={save} />)

    expect(screen.queryByTestId('eds-icon-path')).toHaveAttribute(
      'd',
      save.svgPathData,
    )
  })
  it('Has correct color', () => {
    render(<Icon name="save" color="red" title="icon" />)

    expect(screen.getByLabelText('icon')).toHaveAttribute('fill', 'red')
  })
  it('Has correct size', () => {
    render(<Icon name="save" size={48} title="icon" />)
    const icon = screen.getByLabelText('icon')

    expect(icon).toHaveAttribute('height', '48px')
    expect(icon).toHaveAttribute('width', '48px')
  })
  it('Has correct elements when using title', () => {
    const title = 'Save me'
    render(<Icon name="save" title={title} />)
    const icon = screen.getByLabelText(title)

    expect(icon).toHaveAttribute('aria-labelledby')
    expect(icon).toHaveAttribute('role', 'img')
    expect(screen.queryAllByTitle(title)).toHaveLength(2)
  })
  it('Can extend the css for the component', () => {
    render(<StyledIcon name="save" title="icon" />)
    const icon = screen.getByLabelText('icon')
    expect(icon).toHaveStyleRule('position', 'relative')
    expect(icon).toHaveStyleRule('height', '100px')
    expect(icon).toHaveStyleRule('width', '100px')
  })
})
