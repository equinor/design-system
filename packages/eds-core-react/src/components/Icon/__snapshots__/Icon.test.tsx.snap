// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Matches snapshot using data 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class=""
    fill="currentColor"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class=""
      clip-rule="evenodd"
      d="M5 3h12l4 4v12c0 1.1-.9 2-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2Zm14 16V7.83L16.17 5H5v14h14Zm-7-7c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3Zm3-6H6v4h9V6Z"
      data-testid="eds-icon-path"
      fill-rule="evenodd"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon Matches snapshot using name  1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class=""
    fill="currentColor"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class=""
      clip-rule="evenodd"
      d="M5 3h12l4 4v12c0 1.1-.9 2-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2Zm14 16V7.83L16.17 5H5v14h14Zm-7-7c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3Zm3-6H6v4h9V6Z"
      data-testid="eds-icon-path"
      fill-rule="evenodd"
    />
  </svg>
</DocumentFragment>
`;
